src/test/java/com/parkingapp/parkingservice/infrastructure/database/JdbcVehiclesRepositoryIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+        assertThat(saveResult).isTrue();
+    }
+
+    @Test
+    void shouldSaveVehicleWhenSamePlateAndDifferentUserIdExists() {
+        // GIVEN
+        Vehicle newVehiclePlateAlreadyExists = new Vehicle(
+                UUID.randomUUID(),
+                brand,
+                model,
+                color,
+                plate,
+                country,
+                UUID.randomUUID()
+        );
+
+        // WHEN
+        boolean saveResult = vehicleRepository.saveVehicle(newVehicle);
+        boolean saveResult2 = vehicleRepository.saveVehicle(newVehiclePlateAlreadyExists);
+
+        // THEN
+        Optional<Vehicle> result1 = vehicleRepository.getVehicleByUserIdAndPlate(
+                newVehicle.getUserId(),
+                newVehicle.getPlate()
+        );
+        assertThat(result1).isEqualTo(Optional.of(newVehicle));
+        assertThat(saveResult).isTrue();
+
+        Optional<Vehicle> result2 = vehicleRepository.getVehicleByUserIdAndPlate(
+                newVehiclePlateAlreadyExists.getUserId(),
+                newVehiclePlateAlreadyExists.getPlate()
+        );
+        assertThat(result2).isEqualTo(Optional.of(newVehiclePlateAlreadyExists));
+        assertThat(saveResult2).isTrue();
+    }
+
+    @Test
+    void shouldSaveVehicleWhenSameUserIdAndDifferentPlateExists() {
+        // GIVEN
+        Vehicle newVehicleUserIdAlreadyExists = new Vehicle(
+                UUID.randomUUID(),
+                brand,
+                model,
+                color,
+                "1111XXX",
+                country,
+                userId
+        );
+
+        // WHEN
+        boolean saveResult = vehicleRepository.saveVehicle(newVehicle);
+        boolean saveResult2 = vehicleRepository.saveVehicle(newVehicleUserIdAlreadyExists);
+
+        // THEN
+        Optional<Vehicle> result1 = vehicleRepository.getVehicleByUserIdAndPlate(
+                newVehicle.getUserId(),
+                newVehicle.getPlate()
+        );
+        assertThat(result1).isEqualTo(Optional.of(newVehicle));
+        assertThat(saveResult).isTrue();
+
+        Optional<Vehicle> result2 = vehicleRepository.getVehicleByUserIdAndPlate(
+                newVehicleUserIdAlreadyExists.getUserId(),
+                newVehicleUserIdAlreadyExists.getPlate()
+        );
+        assertThat(result2).isEqualTo(Optional.of(newVehicleUserIdAlreadyExists));
+        assertThat(saveResult2).isTrue();
+    }
+
+    @Test
+    void shouldNotSaveAVehicleWhenPlateAndUserIdExists() {
+        // GIVEN
+        Vehicle duplicateVehicle = new Vehicle(
+                UUID.randomUUID(),
+                brand,
+                model,
+                color,
+                plate,
+                country,
+                userId
+        );
+
+        // WHEN
+        boolean saveResult = vehicleRepository.saveVehicle(newVehicle);
+        boolean saveResult2 = vehicleRepository.saveVehicle(duplicateVehicle);
+
+        // THEN
```

Comment:
fetch user vehicles and check there's only 1 saved and it's the first one saved

Reply:
this is to check that only 1 vehicle is saved per user id and plate and that the vehicle with same user id and plate is not wrongly updated

Reply:
done! 💯